HoldingObjectsLab/PhoneBookRefactored/src/main/java/PhoneBook.java

```diff
@@ -0,0 +1,65 @@
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
+/**
+ * Created by eugenevendensky on 2/7/17.
+ */
+public class PhoneBook {
+
+    public TreeMap<String, Entry> entries = new TreeMap<String, Entry>();
+
+    public void addEntries(String name, String number) {
+        if (entries.containsKey(name)) {
+            Entry temp = entries.get(name);
+            temp.addPhoneNumber(number);
+        } else {
+            Entry entry = new Entry(number);
+            entries.put(name, entry);
+        }
+
+    }
+
+    public String lookUp(String name) {
+        Entry temp = entries.get(name);
+        String numbers = temp.getNumbers();
+        return numbers;
+    }
+
+    public String reverseLookUp(String number) {
+
+        Set set = entries.entrySet();
```

Comment:
You're using raw types here. if you declare this as a `Set<Map.Entry<String, Entry>>` you won't have to do all that typecasting on lines 37, 38, 42 etc.